client-programs/fault-proof/src/main.rs

```diff
@@ -39,9 +43,20 @@ fn main() {
             // and in memory oracle.
             if #[cfg(target_os = "zkvm")] {
                 println!("cycle-tracker-start: boot-load");
-                let raw_boot_info = sp1_zkvm::io::read::<RawBootInfo>();
-                sp1_zkvm::io::commit_slice(&raw_boot_info.abi_encode());
-                let boot: Arc<BootInfo> = Arc::new(raw_boot_info.into());
+                let boot_info_with_no_config = sp1_zkvm::io::read::<BootInfoWithNoConfig>();
+                let rollup_config_bytes = sp1_zkvm::io::read_vec();
+                let boot_info_with_hashed_config = BootInfoWithHashedConfig::new(&boot_info_with_no_config, &rollup_config_bytes);
+                sp1_zkvm::io::commit::<BootInfoWithHashedConfig>(&boot_info_with_hashed_config);
```

Comment:
Did you run into issues having a Vec<u8> inside of `BootInfoWithHashedConfig`? You should just be able to read this in as one struct.

Reply:
Done.